whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -332,7 +329,24 @@ private static Autnum createAutnumResponse(final RpslObject rpslObject) {
         final Autnum autnum = new Autnum();
         autnum.setHandle(rpslObject.getKey().toString());
         autnum.setName(rpslObject.getValueForAttribute(AttributeType.AS_NAME).toString().replace(" ", ""));
-        autnum.setType("DIRECT ALLOCATION");
+        autnum.setType(DIRECT_ALLOCATION);
+        autnum.getEntitySearchResults().addAll(createContactEntities(rpslObject));
+        return autnum;
+    }
+
+    private static Autnum createAsBlockResponse(final RpslObject rpslObject) {
+        final Autnum autnum = new Autnum();
+
+        final String key = rpslObject.getValueForAttribute(AttributeType.AS_BLOCK).toString();
+        final String handle = key.split("-")[0].replace(" ", "");
+        final AsBlockRange blockRange = getAsBlockRange(key);
+
+        autnum.setHandle(handle);
+        //TODO :check what should be the name
+        autnum.setName(key.replace(" ", ""));
```

Comment:
Use trim() instead of replace.. ?

Reply:
Or are there internal spaces you also want to replace?

Reply:
Internal spaces, as CreateAutnumResponse also removes internal spaces.
I believe its because in database we always store the block with space (AS0 - AS6) an while returning JSON we remove the spaces

Reply:
Better not to re-format the key here, but also use String.format("...") with the parsed blockRange values instead. 

So you leave the parsing to AsBlockRange, and you specify exactly how to format the handle.